jetty-start/src/main/java/org/eclipse/jetty/start/Main.java

```diff
@@ -441,7 +441,7 @@ public void start(StartArgs args) throws IOException, InterruptedException
         {
             CommandLineBuilder cmd = args.getMainArgs(args.getDryRunParts());
             cmd.debug();
-            System.out.println(cmd.toQuotedString());
+            System.out.println(cmd);
```

Comment:
@joakime Sorry but I'm not follow the pronouns... what "it" needs to be removed?  what "this" needs to be fixed?
Can you and @lachlan-roberts get together and come up with a grander scale fix for this.  Good point that our ini files are already expanded, so that any quotes in them are literal quotes not shell quotes.    We need to be very clear about all our spaces if they are expanded or not... and the bottom line is that the output of --dry-run needs to be able to be executed by a shell or by a shell script.

Reply:
This is bigger than `--dry-run` alone ...

Use Cases (A): --dry-run output use/consumption

1. Executed by unix shell directly (eg: `exec $ARGS`)
2. Streamed into executable / pipe in unix via args (eg: `xargs`)
3. Used as arguments in java daemon
4. Used as arguments in linux systemd startup metadata
5. Used as arguments in windows daemon
6. Used in `jetty.sh`

Use Cases (B): Quoting of arguments by start.jar

1. From command line
2. From start.d/*.ini with quotes
3. From start.d/*.ini without quotes
4. From module declaration with quotes
5. From module declarattion without quotes

Use Cases (C): How the arguments are consumed

1. The output of `--dry-run`
2. Direct `main.class` invoke of `main(String[] args)`
3. Use in `java.lang.ProcessBuilder(List<String> args)`

The change that lachlan did in this PR ...

* Addressed use cases A1, A2, A6, B1, B2, B4, C1
* But Broke use cases A3, A4, A5, B3+C2, B3+C3, B5+C2, B5+C3

Reply:
@joakime @lachlan-roberts 
> This is bigger than `--dry-run` alone ...

Agreed.  Rather than arbitrarily going through start.jar adding or removing quotes, we need to step back and be very rigorous in understanding what spaces we have and if they are quoted or unquoted.  

> Use Cases (A): --dry-run output use/consumption
> 
> 1. Executed by unix shell directly (eg: `exec $ARGS`)

Obviously this is a quoted valued, so that it can be cut and pasted on the command line and executed.


> 2. Streamed into executable / pipe in unix via args (eg: `xargs`)

ditto

> 3. Used as arguments in java daemon

ditto

> 4. Used as arguments in linux systemd startup metadata

ditto

> 5. Used as arguments in windows daemon

Interesting... are windows command line quoting conventions the same as unix?

> 6. Used in `jetty.sh`

This is also a quoted value.


I.e. The output of --dry-run should be quoted as necessary in all circumstances so that it can be correctly interpreted by a shell.



> 
> Use Cases (B): Quoting of arguments by start.jar

I think this is the more interesting area and probably where our problems lie:

> 1. From command line

The command line is expanded by the shell, thus the arguments that we receive are unquoted.

> 2. From start.d/*.ini with quotes

Our ini files are line separated, so they do not need to be quoted just for separation and preservation of spaces.   So is this an unquoted space?

However, we do expand `${name}` variables in our ini-files, so do we need an escape if an argument needs to have a value that actually contains `${`?  If we need an escape, then this perhaps a quoted space?   Or some hybrid where we assume every line is preceded and terminated with a `"`?

> 3. From start.d/*.ini without quotes

As above.

> 4. From module declaration with quotes

Lines from mod files are copied directly to be lines in ini files, so whatever space ini files are in, then mod file lines are the same.

> 5. From module declarattion without quotes

As above.

So I do not think our issue is with the output of --dry-run, as that simply must be correctly quoted so that a shell understands the argument separators and values.
I think our issue is with our inputs and determining if they are or are not quoted.

Specifically for this particular issue:
 + we have an input of `-Dprop="\"foo bar\""` given on the command line.  this is a quoted value
 + this is after the `-jar` argument so it is passed to the main method as an argument with value `prop="foo bar"`, which is unquoted
 + The output of the `--dry-run` requotes this (as it should) to be `'-Dprop="foo bar"'`.   I'm not seeing the problem with this?   Perhaps it would be better to quote name and value separately to `-Dprop=`"foo bar"'`, but the value should be the same..... ok so I'm not understanding part of the original problem..... I'll experiment some more before commenting more.